### PR TITLE
Add orbit state vectors

### LINKED ISSUE #3: Satellite Location
Hi there,

Users request information related to the satellite location (altitude, latitude, longitude). It would be useful if it is included in the Satellite Extension Specification-

Many thanks and best regards,
Tony

### PR DESCRIPTION
**Pull request by emmanuelmathot**
This PR adds the possibility to provides with orbit state vectors either inline or with a linked file.

It fixes #3

### CODE REVIEW

**Comment by m-mohr** [comment]
I'm wondering whether they schould generally be externalized to keep things small? Just thinking out loud right now.

File: json-schema/schema.json [hunk 1]
```diff
@@ -149,6 +154,20 @@
         "sat:orbit_cycle": {
           "type": "integer",
           "minimum": 1
+        },
+        "sat:orbit_state_vectors": {
+          "type": "object",
+          "patternProperties": {
+            "^.*$": {
```
Comment (m-mohr) [inline json-schema/schema.json:161]: Either use additionalProperties or provide a better regexp for a datetime if that's always the expected value.

**Comment by emmanuelmathot** [comment]
Validation fails with STAC Node Validator while it works fine with this tool: https://jsonschema.dev/
@m-mohr any idea?

File: json-schema/schema.json [hunk 2]
```diff
@@ -149,6 +154,21 @@
         "sat:orbit_cycle": {
           "type": "integer",
           "minimum": 1
+        },
+        "sat:orbit_state_vectors": {
+          "type": "object",
+          "$comment": "The key is a date-time string, and the value is an array of numbers.",
+          "patternProperties": {
+            "^([\\+-]?\\d{4}(?!\\d{2}\b))((-?)((0[1-9]|1[0-2])(\\3([12]\\d|0[1-9]|3[01]))?|W([0-4]\\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\\d|[12]\\d{2}|3([0-5]\\d|6[1-6])))([T\\s]((([01]\\d|2[0-3])((:?)[0-5]\\d)?|24\\:?00)([\\.,]\\d+(?!:))?)?(\\17[0-5]\\d([\\.,]\\d+)?)?([zZ]|([\\+-])([01]\\d|2[0-3]):?([0-5]\\d)?)?)?)?$": {
```
Comment (m-mohr) [inline json-schema/schema.json:162]: Various comments:

- The `\b` might cause the issue with esacping in the node validator. Try `\\b`.
- STAC generally requires `Z`/UTC as timezone so we can probably simplify from `([zZ]|([\\+-])([01]\\d|2[0-3]):?([0-5]\\d)?)?` to `([zZ])?`
- What is this part about: `W([0-4]\\d`? I've never ween a W in ISO date-times. I guess that's for `W([0-4]\\d`, which is not valid here.

**Comment by m-mohr** [comment]
Before we get too deep into regexes: Would it be good enough to just define a relation type? Is state vectors something that you'd search for? I don't quite like embedding these "big" constructs to STAC such as state vectors, histograms, etc.

**Comment by AleMScof** [comment]
Do you have any updates on this PR? 
We are interested in embedding orbit state vector information for AIS data.

File: README.md [hunk 3]
```diff
@@ -21,14 +21,15 @@ It will often be combined with other extensions that describe the actual data, s
...
+| sat:relative_orbit                    | integer               | The relative orbit number at the time of acquisition.                                                                                                                                                   |
+| sat:anx_datetime                      | string                | The [Ascending Node](https://en.wikipedia.org/wiki/Orbital_node) Crossing (ANX) time, in UTC. It is formatted according to [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6).    |
+| sat:orbit_cycle                       | integer               | The number of repeat cycle done by the satellite at the time of the acquisition. [Repeat cycle](https://ltb.itc.utwente.nl/page/498/concept/81577) is the time between two successive identical orbits. |
+| sat:orbit_state_vectors               | Map<string, double[]> | The state vectors of the satellite at the time of acquisition.                                                                                                                                          |
```
Comment (m-mohr) [inline README.md:32]:
```suggestion
| sat:orbit_state_vectors               | Map<string, \[number]> | The state vectors of the satellite at the time of acquisition.                                                                                                                                          |
```

File: README.md [hunk 4]
```diff
@@ -73,6 +74,60 @@ The repeat cycle of the satellite. The repeat cycle is the number of orbits requ
...
+      4590328.5904560406,
+      5114.1961953273549,
+      -295.79384304587683,
+      -5604.2108611458289
+    ]
+  }
+}
+```
+
+Optionaly, the state vectors can be provided in a separate file and referenced as a link with relationship type `sat:osv`.
```
Comment (m-mohr) [inline README.md:120]:
```suggestion
It is recommended that the state vectors are provided in a separate file and referenced as a link with relationship type `sat:osv`. The file should correspond to the example above.
```

**Review by m-mohr — APPROVED** [review]

**Pull request merged**